prover/crates/bin/prover_autoscaler/src/global/queuer.rs

```diff
@@ -39,11 +50,16 @@ impl Queuer {
             .json::<Vec<VersionedQueueReport>>()
             .await
             .context("Failed to read response as json")?;
-        Ok(Queue {
-            queue: response
-                .iter()
-                .map(|x| (x.version.to_string(), x.report.prover_jobs.queued as u64))
-                .collect::<HashMap<_, _>>(),
-        })
+        Ok(response
+            .iter()
+            .flat_map(|x| {
+                jobs.iter().map(move |j| {
```

Comment:
I need to iterate over jobs because I can't iterate over struct fields. Instead of providing list of jobs from the arguments, I could just hardcode all of them here and parse all of them.

Not parsing them per job would make it harder for the caller.